packages/kbn-apm-synthtrace/src/scenarios/degraded_logs.ts

```diff
@@ -23,45 +39,45 @@ const scenario: Scenario<LogDocument> = async (runOptions) => {
     generate: ({ range, clients: { logsEsClient } }) => {
       const { logger } = runOptions;
 
-      // Logs Data logic
-      const MESSAGE_LOG_LEVELS = [
-        { message: 'A simple log', level: 'info' },
-        {
-          message: 'Another log message',
-          level: 'debug',
-        },
-        { message: 'Error with certificate: "ca_trusted_fingerprint"', level: 'error' },
-      ];
-      const CLOUD_PROVIDERS = ['gcp', 'aws', 'azure'];
-      const CLOUD_REGION = ['eu-central-1', 'us-east-1', 'area-51'];
-
-      const CLUSTER = [
-        { clusterId: generateShortId(), clusterName: 'synth-cluster-1' },
-        { clusterId: generateShortId(), clusterName: 'synth-cluster-2' },
-        { clusterId: generateShortId(), clusterName: 'synth-cluster-3' },
-      ];
+      const constructLogsCommonData = () => {
+        const index = Math.floor(Math.random() * 3);
+        const serviceName = getServiceName(index);
+        const logMessage = MESSAGE_LOG_LEVELS[index];
+        const cluster = getCluster(index);
+        const cloudRegion = getCloudRegion(index);
 
-      const SERVICE_NAMES = Array(3)
-        .fill(null)
-        .map((_, idx) => `synth-service-${idx}`);
+        return {
+          index,
+          serviceName,
+          logMessage,
+          cluster,
+          cloudRegion,
+        };
+      };
 
       const datasetSynth1Logs = (timestamp: number) => {
-        const index = Math.floor(Math.random() * 3);
+        const {
+          serviceName,
+          logMessage: { level, message },
+          cluster: { clusterId, clusterName },
+          cloudRegion,
+        } = constructLogsCommonData();
+
         return log
           .create({ isLogsDb })
           .dataset('synth.1')
-          .message(MESSAGE_LOG_LEVELS[index].message as string)
-          .logLevel(MESSAGE_LOG_LEVELS[index].level)
-          .service(SERVICE_NAMES[index])
+          .message(message as string)
```

Comment:
Why would we need to cast that to `string` here (and further below)? Isn't it always a string already?